Cargo.toml

```diff
@@ -69,9 +69,12 @@ members = [
 [profile.release]
 overflow-checks = true
 
-[target.'cfg(all(not(target_env = "msvc"), not(target_os="macos")))'.dependencies]
+[target.'cfg(all(not(target_env = "msvc"), not(target_os="macos"), not(feature = "profiling")))'.dependencies]
 jemallocator = { version = "0.3.0" }
+[target.'cfg(all(not(target_env = "msvc"), not(target_os="macos"), feature = "profiling"))'.dependencies]
```

Comment:
The doc below mentioned that this feature is only available on Linux, so why not test target_os is Linux directly? Does the config mean that toolchain that is not using msvc(e.g: mingw) will work on Windows?

Reply:
Beats me! I'm confused, too.

I just keep the same with `Cargo.toml`.
https://github.com/nervosnetwork/ckb/blob/dad394ea3f50f518a40e5a8a457dfb6811ba245a/Cargo.toml#L72-L73

This two lines are introduced by #1799. Maybe @jjyr could answer this.

Reply:
I think I just copied it from somewhere, can't find the first place.

After checking the `jemallocator` crate, I belieave we should use the `cfg(all(target_arch = "x86_64",target_os="linux"))` compile condition, which only enable `jemalloc` on the tier1 supports platform.

https://crates.io/crates/jemallocator

Reply:
- It's copied from [README.md of jemallocator](https://github.com/gnzlbg/jemallocator/blob/c27a859e98e3cb790dc269773d9da71a1e918458/README.md#documentation).

  - We copied `target_env = "msvc"`.
  - And we appended `target_os="macos"`.

- [And jemallocator can be compiled on Windows without msvc.](https://github.com/gnzlbg/jemallocator/blob/c27a859e98e3cb790dc269773d9da71a1e918458/README.md#platform-support) So the official `README.md` suggests `target_env = "msvc"`.

  | Windows targets                      | build     | run      | tests           | valgrind     |
  |-------------------------------------|-----------|---------|--------------|--------------|
  | `x86_64-pc-windows-msvc` (tier 1)   | ✗         | ✗       | ✗            | ✗            |
  | `i686-pc-windows-msvc` (tier 1)     | ✗         | ✗       | ✗            | ✗            |
  | `x86_64-pc-windows-gnu` (tier 1)    | ✓         | ✓       | ✗            | ✗            |
  | `i686-pc-windows-gnu` (tier 1)      | ✓         | ✓       | ✗            | ✗            |